docs/sop.rst

```diff
@@ -0,0 +1,135 @@
+different modes. In *local* mode, the Kafka provider deploys a single node Kafka/Zookeeper instance
+inside the cluster and configures it to auto-create the topics. In *operator* mode, the provider
+assumes a Strimzi Kafka instance is present and will create ``KafkaTopic`` CRs to provide the
+topics.  In *app-interface* mode, no resources are deployed and it is assumed app-interface has
+already created the requested topics. For more information on the configuration of each of these
+providers and their modes, please see the links below.
+
+List of providers:
+
+- `Kafka <https://redhatinsights.github.io/clowder/api_reference.html#k8s-api-cloud-redhat-com-clowder-v2-apis-cloud-redhat-com-v1alpha1-databaseconfig>`_
+- `Logging <https://redhatinsights.github.io/clowder/api_reference.html#k8s-api-cloud-redhat-com-clowder-v2-apis-cloud-redhat-com-v1alpha1-loggingconfig>`_
+- `Object storage <https://redhatinsights.github.io/clowder/api_reference.html#k8s-api-cloud-redhat-com-clowder-v2-apis-cloud-redhat-com-v1alpha1-objectstoreconfig>`_
+- `In-memory DB <https://redhatinsights.github.io/clowder/api_reference.html#k8s-api-cloud-redhat-com-clowder-v2-apis-cloud-redhat-com-v1alpha1-inmemorydbconfig>`_
+- `Relational DB <https://redhatinsights.github.io/clowder/api_reference.html#k8s-api-cloud-redhat-com-clowder-v2-apis-cloud-redhat-com-v1alpha1-databaseconfig>`_
+- `Web <https://redhatinsights.github.io/clowder/api_reference.html#k8s-api-cloud-redhat-com-clowder-v2-apis-cloud-redhat-com-v1alpha1-webconfig>`_
+- `Metrics <https://redhatinsights.github.io/clowder/api_reference.html#k8s-api-cloud-redhat-com-clowder-v2-apis-cloud-redhat-com-v1alpha1-metricsconfig>`_
+
+Target Namespace
+^^^^^^^^^^^^^^^^
+
+Clowder will provision two types of resources during operation, *environmental* and *application*
+specific. Application resources will be deployed into the same namespace the ClowdApp resides in.
+Environmental resources, such as the Kafka/Zookeeper from the exmaple in the *Modes* section, will
+be placed in the ``ClowdEnvironment``'s target namespace. This is configured by setting the
+``targetNamespace`` attribute of the ``ClowdEnvironment`` and should always be set for production
+environments. If it is omitted, a random target namespace is generated instead. The name of this
+resource can be found by inspecting the ``status.targetNamespace`` of the ClowdEnvironment resource.
+
+ClowdApp
+++++++++
+
+**abbreviated to ``app`` in k8s**
+
+The ``ClowdApp`` CRD is responsible for configuring an application and is namespace scoped. Any
+resources Clowder creates on behalf of the application will reside in the same namespace that the
+``ClowdApp`` resources is applied to. As such the ``ClowdApp`` name must be unique within a
+particular namespace.  An ``ClowdApp`` does not have to be placed in the ``ClowdEnvironment``'s
+target namespace.
+
+A ``ClowdApp`` may define multiple services inside it. These services, though defined by a
```

Comment:
You refer to the `pod` specs as "services", causing an inconsistency.  Is that what we want to call them?

Reply:
I'm not sure, should we do a spec change to bring the terminology inline before this is used outside of the platform?

Reply:
Yeah but we should make sure we choose a good name, which is a challenge.

Reply:
Agreed......